doc/source/profiling.md

```diff
@@ -0,0 +1,76 @@
+# Profiling
+
+xDEM has a built-in profiling tool, that can be used to provide more insight on the memory and time use of function.
```

Comment:
Perfect intro sentence. :slightly_smiling_face: 
Small typo at the end: "of a function".